one-ui/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -1,30 +1,47 @@
-import React, { CSSProperties, useState } from 'react';
+import React, { CSSProperties, useState, useEffect } from 'react';
 import styles from './mapping-card.module.scss';
 import {Card, Icon, Tooltip, Row, Col, Modal} from 'antd';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import {faTrashAlt} from '@fortawesome/free-regular-svg-icons';
 import sourceFormatOptions from '../../../config/formats.config';
-import { convertDateFromISO } from '../../../util/conversionFunctions';
+import { convertDateFromISO, getInitialChars } from '../../../util/conversionFunctions';
 import CreateEditMappingDialog from './create-edit-mapping-dialog/create-edit-mapping-dialog';
+import SourceToEntityMap from './source-entity-map/source-to-entity-map';
+import {getResultsByQuery, getDoc} from '../../../util/search-service'
 
 interface Props {
     data: any;
     entityName: any;
+    getMappingArtifactByMapName: any;
     deleteMappingArtifact: any;
     createMappingArtifact: any;
+    updateMappingArtifact: any;
     canReadOnly: any;
     canReadWrite: any;
+    entitiesInfo: any;
   }
 
 const MappingCard: React.FC<Props> = (props) => {
     const [newMap, setNewMap] = useState(false);
     const [title, setTitle] = useState('');
     const [mapData, setMapData] = useState({});
+    const [mapName, setMapName] = useState({});
     const [dialogVisible, setDialogVisible] = useState(false);
     const [loadArtifactName, setLoadArtifactName] = useState('');
+    const [mappingVisible, setMappingVisible] = useState(false);
+    const [sourceData, setSourceData] = useState<any[]>([]);
+    const [sourceURI,setSourceURI] = useState('');
+    const [docNotFound, setDocNotFound] = useState(false);
+
+    //For Entity table
+    const [entityData, setEntityData] = useState<any[]>([]);
```

Comment:
The user is really constructing a mapping artifact, not an entity. Can we instead say there's a "source table" and a "mapping table"?

Reply:
@rjrudin - I used the name 'EntityData' here because it seems consistent with the UI as we have been calling the right-hand side table in mapping UI as Entity table only.